src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -366,31 +366,44 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
     /*******************************************************************************************************
     * @description utility to return the current StartDate or EndDate of a Seasonal Address
     * @param addr the Address to look at
-    * @param fStart whether to return the StartDate or EndDate
+    * @param returnStartDate whether to return the StartDate or EndDate
     * @return Date
     */ 
-    private static Date dtStartOrEndOfAddrSeasonal(Address__c addr, boolean fStart) {    
+    private static Date getStartOrEndDateOfAddrSeasonal(Address__c addr, boolean returnStartDate) {    
+        integer startYear = integer.valueOf(addr.Seasonal_Start_Year__c);
         integer startMonth = integer.valueOf(addr.Seasonal_Start_Month__c);
         integer startDay = integer.valueOf(addr.Seasonal_Start_Day__c);
+        
+        integer endYear = integer.valueOf(addr.Seasonal_End_Year__c);
         integer endMonth = integer.valueOf(addr.Seasonal_End_Month__c);
         integer endDay = integer.valueOf(addr.Seasonal_End_Day__c);
-        Date dtStart;
-        Date dtEnd;
+        
+        Date startDate;
+        Date endDate;
         Date dtToday = system.today();
-        //seasonal address is contained within a single year
-        if (startMonth <= endMonth) { 
-            dtStart = date.newInstance(dtToday.year(), startMonth, startDay);
-            dtEnd = date.newInstance(dtToday.year(), endMonth, endDay);
-        //seasonal address overlaps the year, and this month is past the start month
-        } else if (startMonth <= dtToday.month()){
-            dtStart = date.newInstance(dtToday.year(), startMonth, startDay);
-            dtEnd = date.newInstance(dtToday.year()+1, endMonth, endDay);
-        //seasonal address overlaps the year, and this month is before the start month
+        
+        //start and end year have been provided (we don't allow providing only start or only end - addrSeasonalPartialYear error)
+        if(startYear != null && endYear != null) {
+            startDate = date.newInstance(startYear, startMonth, startDay);
+            endDate = date.newInstance(endYear, endMonth, endDay);
+        //start and end year have not been provided
         } else {
-            dtStart = date.newInstance(dtToday.year()-1, startMonth, startDay);
-            dtEnd = date.newInstance(dtToday.year(), endMonth, endDay);
+	        //seasonal address is contained within a single year
+	        if (startMonth <= endMonth && startDay <= endDay) { 
```

Comment:
Thanks for catching this. I made this change because the last check in testSeasonalAddrOverlapLogic was failing. However, what I should have written is what I'm going to commit in just a sec (together with a change that would have caught the issue you mention).